### ISSUE TITLE
getShopReceipts: Proposed design change for getShopReceipts for post-sale activities

### ISSUE THREAD

**Issue by Gareth064** [etsy/open-api#604]
### Is there an existing issue for this?

- [X] I have searched the existing tickets

### Current Endpoint Design

When using the `min_last_modified` & `max_last_modified` query parameters on this endpoint, the response **does not** include receipts where the following activities have happened within the stated parameters timespan.

- Receipt Cancelled
- Receipt Refunded Fully (Not Cancelled)
- Receipt Refunded Partially
- Expected Ship\Dispatch Date on Receipt Changed

### Proposed Endpoint Design Change

When using the `min_last_modified` & `max_last_modified` query parameters on this endpoint, the response **_should_** include receipts where the following activities have happened within the stated parameters timespan.

- Receipt Cancelled
- Receipt Refunded Fully (Not Cancelled)
- Receipt Refunded Partially
- Expected Ship\Dispatch By Date on Receipt Changed

**Other possibilities**

1. Add Query Params like `was_cancelled`, `was_fullyrefunded`, `was_partiallyrefunded`. Not sure how to handle the Ship\Dispatch date with a boolean.
2. Add brand new endpoints for these very specific activities.

### Why are you proposing this change?

The following issues were considered closed because they said the `updated_timestamp` on a Receipt was not updated to reflect post-sale activities. And you "fixed" those bugs via [3.0.0-general-release-2022-07-19](https://github.com/etsy/open-api/releases/tag/3.0.0-general-release-2022-07-19) where you replace the `updated_timestamp` in the "response" with the latest updated_timestamp from related objects in the Receipt.

Previous issues: https://github.com/etsy/open-api/issues/513, https://github.com/etsy/open-api/issues/339, https://github.com/etsy/open-api/issues/232

But what we all wanted and expected, was yes, the updated_timestamp on the Receipt reflected that things have happened post-sale, but **also** that we can use the `min_last_modified` & `max_last_modified` query parameters on the [getShopReceipts ](https://developers.etsy.com/documentation/reference/#operation/getShopReceipts)endpoint.

Based on the previous issues raised about this, these are the post-sale activities which we expect to be able to pick up using the getShopReceipts endpoint along with the `min_last_modified` & `max_last_modified` query parameters.

- Receipt Cancelled
- Receipt Refunded Fully (Not Cancelled)
- Receipt Refunded Partially
- Expected Ship\Dispatch Date on Receipt Changed

**Comment by chrisrogala** [comment]
I completely agree with this. By the end of next month, we will have over 1000 orders in our queue. Currently, the only way to identify if the order has been cancelled is if we sent a GET request to check the status on those 1000 orders individual. There is nothing in the documentation that states this endpoint would exclude these Receipts, so I would think they would be included.

**Comment by etsyachristensen** [comment]
This has been logged for discussion in our group.

**Comment by sowsinsk** [comment]
These are great suggestions, and would be helpful to us as well.

**Comment by KushGoyal** [comment]
I really need this feature too. Finding out cancelled orders is a huge headache.

**Comment by maurasz** [comment]
hello, any updates regarding this issue?
